task.sql

```diff
@@ -4,30 +4,66 @@
 USE ShopDB;
 
 CREATE TABLE Countries (
-    ID INT,
+    ID INT AUTO_INCREMENT,
     Name VARCHAR(50),
     PRIMARY KEY (ID)
 );
 
+CREATE TABLE Cities (
+		ID INT AUTO_INCREMENT,
+		Name VARCHAR(50),
+		CountryID INT,
+		FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
+		PRIMARY KEY (ID)
+);
+
+CREATE TABLE Addresses (
+		ID INT AUTO_INCREMENT,
+		StreetName VARCHAR(100),
+		CityID INT,
+		FOREIGN KEY (CityID) REFERENCES Cities(ID) ON DELETE NO ACTION,
+		PRIMARY KEY (ID)
+);
+
+CREATE TABLE Warehouses (
+		ID INT AUTO_INCREMENT,
+		Name VARCHAR(50),
+		AddressID INT,
+		FOREIGN KEY (AddressID) REFERENCES Addresses(ID) ON DELETE NO ACTION,
```

Comment:
For the foreign key constraint on `AddressID`, consider using `ON DELETE CASCADE` if you want to automatically delete dependent warehouses when an address is deleted.